odl/trafos/backends/pyfftw_bindings.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Bindings to the ``pyFFTW`` backend for Fourier transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import range
+from future.utils import raise_from
+
+from multiprocessing import cpu_count
+import numpy as np
+try:
+    import pyfftw
+    PYFFTW_AVAILABLE = True
+except ImportError:
+    PYFFTW_AVAILABLE = False
+
+from odl.util import is_real_dtype, dtype_repr, TYPE_MAP_R2C
+
+
+__all__ = ('pyfftw_call', 'PYFFTW_AVAILABLE')
+
+
+def pyfftw_call(array_in, array_out, direction='forward', axes=None,
+                halfcomplex=False, **kwargs):
+    """Calculate the DFT with pyfftw.
+
+    The discrete Fourier (forward) transform calcuates the sum::
+
+        f_hat[k] = sum_j( f[j] * exp(-2*pi*1j * j*k/N) )
+
+    where the summation is taken over all indices
+    ``j = (j[0], ..., j[d-1])`` in the range ``0 <= j < N``
+    (component-wise), with ``N`` being the shape of the input array.
+
+    The output indices ``k`` lie in the same range, except
+    for half-complex transforms, where the last axis ``i`` in ``axes``
+    is shortened to ``0 <= k[i] < floor(N[i]/2) + 1``.
+
+    In the backward transform, sign of the the exponential argument
+    is flipped.
+
+    Parameters
+    ----------
+    array_in : `numpy.ndarray`
+        Array to be transformed
+    array_out : `numpy.ndarray`
+        Output array storing the transformed values, may be aligned
+        with ``array_in``.
+    direction : {'forward', 'backward'}
+        Direction of the transform
+    axes : int or `sequence` of ints, optional
+        Dimensions along which to take the transform. ``None`` means
+        using all axis and is equivalent to ``np.arange(ndim)``.
+    halfcomplex : bool, optional
+        If ``True``, calculate only the negative frequency part along the
+        last axis. If ``False``, calculate the full complex FFT.
+        This option can only be used with real input data.
+
+    Other Parameters
+    ----------------
+    fftw_plan : ``pyfftw.FFTW``, optional
```

Comment:
I'll try it, but for the given examples, `intersphinx` downloads a very specific `objects.inv` file from the given URI, and I'm not sure `pyfftw` has such a thing. It's very hard to google also.